pkg/auth/providers/azure/azure_provider.go

```diff
@@ -354,20 +355,26 @@ func (ap *azureProvider) userGroupsToPrincipals(
 	azureClient *azureClient,
 	groups graphrbac.UserGetMemberGroupsResult,
 ) ([]v3.Principal, error) {
-	var groupPrincipals []v3.Principal
-	for _, group := range *groups.Value {
-		groupObj, err := azureClient.groupClient.Get(context.Background(), group)
-		if err != nil {
-			return groupPrincipals, err
-		}
+	var g errgroup.Group
+	groupPrincipals := make([]v3.Principal, len(*groups.Value))
+	for i, group := range *groups.Value {
+		g.Go(func() error {
+			groupObj, err := azureClient.groupClient.Get(context.Background(), group)
```

Comment:
@dramich is the `azureClient.groupClient` thread-safe?

Reply:
Yes... it's a big ugly wrapper around http.Client.